lib/plugins/aws/deployFunction.js

```diff
@@ -63,6 +65,35 @@ class AwsDeployFunction {
     }
   }
 
+  // TODO: TESTS
+  checkIfFunctionChangesBetweenImageAndHandler() {
+    const functionObject = this.serverless.service.getFunction(this.options.function);
+    const remoteFunctionPackageType = this.serverless.service.provider.remoteFunctionData
+      .Configuration.PackageType;
+
+    if (functionObject.handler && remoteFunctionPackageType === 'Image') {
+      const errorMessage = [
+        `The function "${this.options.function}" you want to update with handler was previously packaged as an image.`,
+        ' Please run "serverless deploy" to ensure consistent deploy.',
+      ];
+      throw new this.serverless.classes.Error(
```

Comment:
To maintain single convention it'll be nice to always rely on `ServerlessError` as  required from `lib/servelress-error` (altough I know that probably other errors in this file are constructed via `classes.Error` - I've opened an issue to cleanup all such instances: https://github.com/serverless/serverless/issues/8783)